src/main/java/io/cdap/plugin/google/common/GoogleDriveClient.java

```diff
@@ -77,31 +84,40 @@ protected Credential getCredentials() throws IOException {
     switch (authType) {
       case OAUTH2:
         credential = new GoogleCredential.Builder()
-          .setTransport(httpTransport)
-          .setJsonFactory(JSON_FACTORY)
-          .setClientSecrets(config.getClientId(),
-                            config.getClientSecret())
-          .build();
+            .setTransport(httpTransport)
+            .setJsonFactory(JSON_FACTORY)
+            .setClientSecrets(config.getClientId(),
+                config.getClientSecret())
+            .build();
         credential.setRefreshToken(config.getRefreshToken());
         break;
       case SERVICE_ACCOUNT:
         if (config.isServiceAccountJson()) {
-          InputStream jsonInputStream = new ByteArrayInputStream(config.getServiceAccountJson().getBytes());
+          InputStream jsonInputStream = new ByteArrayInputStream(
+              config.getServiceAccountJson().getBytes());
           credential = GoogleCredential.fromStream(jsonInputStream);
-        } else if (config.isServiceAccountFilePath() && !Strings.isNullOrEmpty(config.getServiceAccountFilePath())) {
-          credential = GoogleCredential.fromStream(new FileInputStream(config.getServiceAccountFilePath()));
-        } else {
-          credential = GoogleCredential.getApplicationDefault(httpTransport, JSON_FACTORY);
-        }
-        break;
+          break;
+        } else if (config.isServiceAccountFilePath() && !Strings.isNullOrEmpty(
+            config.getServiceAccountFilePath())) {
+          credential = GoogleCredential.fromStream(
```

Comment:
Should we uniformly use GoogleCredentials here too? That way this will work properly in hybrid mode too and simplify the code a bit.

Reply:
ack, done.